pallets/utxo/src/lib.rs

```diff
@@ -1079,20 +1090,13 @@ where
         data: &Vec<u8>,
         outpoints: &Vec<H256>,
     ) -> Result<(), DispatchError> {
-        spend::<T>(
-            caller,
-            &Transaction {
-                inputs: coin_picker::<T>(outpoints)?,
-                outputs: vec![TransactionOutput::new_call_pp(
-                    value,
-                    dest.clone(),
-                    true,
-                    data.clone(),
-                )],
-                time_lock: Default::default(),
-            },
-        )
-        .map_err(|_| "Failed to spend the transaction!")?;
+        let tx = Transaction {
+            inputs: construct_inputs::<T>(outpoints)?,
+            outputs: vec![TransactionOutput::new_call_pp(value, dest.clone(), true, data.clone())],
```

Comment:
Not for this PR and not a major issue either. In general I don't like having `bool` as a function parameter since it is not obvious what it is for at the call site. Maybe a wee custom enum like `enum FundTx { Yes, No }` would make things cleaner? It is a bit more involved to use it though.